### PR TITLE
chapter 6

### PR DESCRIPTION
**Pull request by chan227**

### CODE REVIEW

**Comment by chan227** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```
Comment (chungpv-1008) [inline app/models/user.rb]: please put into file settings.yml. Use gem config to create that file(https://github.com/rubyconfig/config). Moreover, the order is constants, associations, validates, callbacks, enum, scope, ...
Remember the line break after each type
Similar to ther number '255', '6', ...

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  validates :email, presence: true, length: { maximum: 255 }, format: { with: VALID_EMAIL_REGEX }, uniqueness: true
```
Comment (chungpv-1008) [inline app/models/user.rb:5]: Move '255' into file settings.yml

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  has_secure_password
+  validates :password, presence: true, length: { minimum: 6 }
```
Comment (chungpv-1008) [inline app/models/user.rb:7]: similar to number '6'

File: app/models/user.rb (hunk 2 above)
Comment (chungpv-1008) [inline app/models/user.rb:5]: Move '255' into file settings.yml

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
  before_save { email.downcase! }
```

File: app/models/user.rb (hunk 3 above)
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
  validates :password, presence: true, length: { minimum: 6 }
  
  has_secure_password
```

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
```
Comment (saobv-2223) [inline app/models/user.rb:1]: Constant variable you put into firstly, validate secondly and last is callback (before_save)

File: app/models/user.rb (hunk 4 above)
Comment (chan227) [reply app/models/user.rb]: Done.

File: app/models/user.rb (hunk 3 above)
Comment (chan227) [reply app/models/user.rb]: Done

File: app/models/user.rb (hunk 5 above)
Comment (chan227) [reply app/models/user.rb:1]: Done

File: app/models/user.rb (hunk 1 above)
Comment (chan227) [reply app/models/user.rb]: done

**Comment by chan227** [comment]
ready

**Comment by vanvtt-0952** [comment]
merged

**Pull request merged**